### ISSUE TITLE
divide by zero warning in handicap_from_score

### ISSUE THREAD

**Issue by shiroyagi** [jatkinson1000/archeryutils#108]
```
>>> import archeryutils as au
>>> au.handicaps.handicap_from_score(420, au.load_rounds.WA_outdoor.wa720_70, "AGB")
python3.11/site-packages/archeryutils/handicaps/handicap_scheme.py:653: RuntimeWarning: divide by zero encountered in scalar divide
  stry = -fcur * (xcur - xpre) / (fcur - xpre)
np.float64(51.43806953277407)
```
The warning does not happen on the second call with the same score (which gives the same result) or with other scores.

**Comment by jatkinson1000** [comment]
Thanks for raising this.

The good news is that it is still returning the correct number for the handicap, but it would be better not to have the warning.

I have looked into it and it occurs when the score being requested is exactly 300 points below the maximum score for a round (as the 300 is arbitrarily used as a starting point in the rootfinding algorithm).
The algorithm is robust enough that it can handle this to produce the correct answer, but that first iteration results in a warning being printed.

I'll think about the nicest (non-hacky) way to resolve this and generate a patch in due course.

On an unrelated note - can I ask what you use this package for?
Whilst I'm now trying to minimise any API changes it is useful to know the size of the userbase and their applications :)

**Comment by jatkinson1000** [comment]
@shiroyagi sounds good, thanks!

I have now merged a fix for this into the main branch - see #109 

This won't make it to PyPI until the next release (likely after other small PRs and the larger #103 are merged) so you can either work with a local install by using:
```
pip install "archeryutils @ git+https://github.com/jatkinson1000/archeryutils.git"
```
or similar, or continue with the PyPI version from pip and live with the warning until the next release.
